Userland/Libraries/LibJS/Module.cpp

```diff
@@ -1,22 +1,120 @@
 /*
  * Copyright (c) 2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2022, David Tuin <davidot@serenityos.org>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <LibJS/CyclicModule.h>
 #include <LibJS/Module.h>
+#include <LibJS/Runtime/ModuleNamespaceObject.h>
 
 namespace JS {
 
-Module::Module(Realm& realm, StringView filename)
-    : m_vm(realm.vm())
-    , m_realm(make_handle(&realm))
-    , m_filename(filename)
+Module::Module(Realm& realm, String filename)
+    : m_realm(make_handle(&realm))
+    , m_filename(move(filename))
 {
 }
 
 Module::~Module()
 {
 }
 
+// 16.2.1.5.1.1 InnerModuleLinking ( module, stack, index ), https://tc39.es/ecma262/#sec-InnerModuleLinking
+ThrowCompletionOr<u32> Module::inner_module_linking(VM& vm, Vector<Module*>&, u32 index)
+{
+    // Note: Until we have something extending module which is not SourceTextModule we crash.
```

Comment:
> Could be a TODO
This (and the one below) are already implemented in SourceTextModule but have to be implemented in the spec for Modules
However currently SourceTextModule is the only non-abstract class so if we end up in this method we did something wrong.

Should I maybe change the comment to be more clear about that?